src/store/token/actions.js

```diff
@@ -203,29 +208,168 @@ export function requestUBIPayout(payout) {
-    },
-    () => {
-      return true;
-    },
-    {},
-    () => {
-      return core.token.updateTransferSteps(from, to, value);
-    },
-  );
+// async function transferLoopHops(
+//   from,
+//   to,
+//   value,
+//   paymentNote,
+//   initialHops = PATHFINDER_HOPS_DEFAULT,
+//   requestedMaxAttempts = 3,
+//   waitAfterFail = 5000,
+// ) {
+//   // Count all attempts to retry with fewer hops when something fails
+//   let attempt = 1;
+//   let hops = initialHops;
+//   const maxAttempts = Math.min(initialHops, requestedMaxAttempts);
+//   const TRIED_TOO_MANY_TIMES = 'Tried too many times waiting for condition.';
+
+//   // Helper method to wait for a few milliseconds before we move on
+//   async function wait(ms) {
+//     return new Promise((resolve) => setTimeout(resolve, ms));
+//   }
+
+//   // eslint-disable-next-line no-constant-condition
+//   while (true) {
+//     try {
+//       // Make transfers request and wait for response
+//       const response = core.token.transfer(from, to, value, paymentNote, hops);
+
+//       // Return and exit function when the there is no error
+//       return response;
+//     } catch (error) {
+//       // Upon error clean the transfer edges - SHOULD WE DO THIS STILL?
+//       // await core.token.updateTransferSteps(from, to, value, hops);
+//       if (attempt >= maxAttempts) {
+//         // We tried too many times
+//         throw error;
+//       }
+
+//       // Wait when request failed to prevent calling the request too often
+//       if (error.message !== TRIED_TOO_MANY_TIMES) {
+//         await wait(waitAfterFail);
+//       }
+
+//       // Lets try again with fewer hops
+//       attempt += 1;
+//       hops -= 1;
+//     }
+//   }
+// }
+
+// async function loopTransfer(
+//   from,
+//   to,
+//   value,
+//   paymentNote,
+//   hops = PATHFINDER_HOPS_DEFAULT,
+// ) {
+//   return await retryLoopUpdateParam(
+//     // request fn
+//     (hops) => {
+//       return core.token.transfer(from, to, value, paymentNote, hops);
+//     },
+//     // loop fn
+//     () => {
+//       return true;
+//     },
+//     {},
+//     // error fn
+//     (hops) => {
+//       return core.token.updateTransferSteps(from, to, value, hops);
+//     },
+//     // param update fn
+//     (param) => {
+//       return param - 1;
+//     },
+//     hops,
+//   );
+// }
+
+async function loopTransfer(
+  from,
+  to,
+  value,
+  paymentNote,
+  hops,
+  attemptsLeft,
+  errorsMessages = '',
+) {
+  /* eslint-disable no-console */
+  if (attemptsLeft === 0 || hops === 0) {
+    // cannot attempt transfer
+    console.log(
+      '-- ran out of tries',
+      { attemptsLeft },
+      { hops },
+      { errorsMessages },
+    );
+    throw new TransferError(errorsMessages);
+  }
+  try {
+    console.log('Attempting with params ', {
+      from,
+      to,
+      value,
+      paymentNote,
+      hops,
+    });
+    return core.token.transfer(from, to, value, paymentNote, hops);
+  } catch (error) {
```

Comment:
I don't seem to be able to catch the 504

Reply:
try now after changing the timeout in the api

Reply:
what was the reason console.log hasn't been displayed? are they now?

Reply:
Yes it changed to 422 but it is catched in core and core throws errors according to here: https://github.com/CirclesUBI/circles-core/blob/b661de206bbaaa3a8aca4cc289e0539c48a8600e/src/token.js#L562